addons/event/static/src/client_action/event_registration_summary_dialog.js

```diff
@@ -82,8 +83,11 @@ export class EventRegistrationSummaryDialog extends Component {
     }
 
     async onRegistrationConfirm() {
-        await this.orm.call("event.registration", "action_set_done", [this.registration.id]);
-        this.registrationStatus.value = "confirmed_registration";
+        if (this.registrationStatus.value !== "confirmed_registration") {
+            this.button.state = false
+            await this.orm.call("event.registration", "action_set_done", [this.registration.id]);
```

Comment:
Need to reenable it on failure otherwise you can softlock the flow :lock: 
```suggestion
            this.button.state = false
            await this.orm.call("event.registration", "action_set_done", [this.registration.id]).catch(() => this.button.state = true);
```